### DISCUSSION TITLE
Select smallest object always when clicked if nested in larger object

### DISCUSSION THREAD

**Discussion by wktdev** [fabricjs/fabric.js#8504] [Q&A] [answered]
**The problem**

1. You have a Rect that is _transparent_ and 500x500. 
2. You have another _**smaller**_ shape placed inside of the Rect and you click away, and then click the smaller shape, the larger outer Rect is always selected. 

I don't want the larger outer shape to be selected.
I want the small shape to always be selected.
More to the point, I want the smaller shapes that are nested inside transparent shapes to take precedence in select-ability. 
In other words, the smaller the object the more its z-index takes precedence. 

How do I do this?

I tried playing with preserveObjectStacking: true but I couldn't get what I was looking for. 

I always want objects that are smaller and encapsulated by larger objects to to take precedence in select-ability. 

Edit:

I found this method for anyone querying this problem. Allows you to create a button, select the shape and move its z index to the top of the pile.

```

let bringToFront = ( )=> {
  let activeObj = canvas.getActiveObject();
  activeObj && canvas.bringToFront(activeObj).discardActiveObject(activeObj).renderAll();
}


```

**Comment by ShaMan123** [comment, marked as answer]
#7670

**Reply by ShaMan123** [reply]
related #8492
